apps/api/src/interview-slot/interview-slot.service.ts

```diff
@@ -6,12 +6,76 @@ import {
 import { InterviewStatus } from '@prisma/client';
 import { PrismaService } from 'src/prisma.service';
 
+import { CreateInterviewSlotDto } from './dto/createInterviewSlot.dto';
+
 @Injectable()
 export class InterviewSlotService {
   constructor(private readonly prisma: PrismaService) {}
 
   async getAll() {
-    const interviewSlots = await this.prisma.interviewSlot.findMany();
+    const interviewSlots = await this.prisma.interviewSlot.findMany({
+      include: {
+        intern: true,
+        interviewers: {
+          include: {
+            interviewer: true,
+            interviewSlot: true,
+          },
+        },
+      },
+    });
+    return interviewSlots;
+  }
+
+  async deleteInterviewSlot(id: string) {
+    const interviewToDelete = await this.prisma.interviewSlot.findUnique({
+      where: { id },
+      include: {
+        intern: true,
+      },
+    });
+
+    if (interviewToDelete.internId) return;
```

Comment:
vrati neki exception tipa BadRequestException('Cannot delete interview with intern on it! Try cancelling interview first'), pa ga prikaži s toast.error